File: modules/heaps/heaps/StudentRecord.java
package heaps;

public class StudentRecord implements Comparable<StudentRecord> {
	
	private String Name;
	private int studentNo;
	private double gpa;
	
	private static int nextNum = 1000;
	
	public StudentRecord(String name, double gpa) {
		super();
		Name = name;
		this.studentNo = nextNum;
		nextNum++;
		this.gpa = gpa;
	}
	
	public String getName() {
		return Name;
	}

	public void setName(String name) {
		Name = name;
	}

	public double getGpa() {
		return gpa;
	}

	public void setGpa(double gpa) {
		this.gpa = gpa;
	}

	@Override
	public String toString() {
		return "[Name=" + Name + ", studentNo=" + studentNo + ", gpa=" + gpa + "]";
	}

	@Override
	public int compareTo(StudentRecord that) {
		return this.Name.compareTo(that.Name);
	}

	@Override
	public boolean equals(Object that) {
		return this.Name.equals(((StudentRecord)that).Name);
	}

}

File: modules/heaps/python/student_record.py
"""
StudentRecord - a comparable element to store in a heap.

1-1 conversion of heaps/StudentRecord.java
(compareTo on the name becomes __lt__ / __eq__ on the name).
"""


class StudentRecord:

    next_num = 1000  # static field shared across all instances

    def __init__(self, name, gpa):
        self.name = name
        self.student_no = StudentRecord.next_num
        StudentRecord.next_num += 1
        self.gpa = gpa

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_gpa(self):
        return self.gpa

    def set_gpa(self, gpa):
        self.gpa = gpa

    def __str__(self):
        return "[Name=" + self.name + ", studentNo=" + str(self.student_no) + ", gpa=" + str(self.gpa) + "]"

    def __lt__(self, that):
        # Orders by name (Java compareTo on the name string).
        return self.name < that.name

    def __eq__(self, that):
        return self.name == that.name
